calculate_rankings.py

```diff
@@ -39,19 +40,40 @@ def main():
     args = parser.parse_args()
 
     date_str = args.date.replace("-", "")
+
     if args.division == "all":
         divisions = DIVISIONS  # Calculate rankings for all divisions at once
     else:
         divisions = [args.division]
 
-    setup_logger(args.output / f"calculate_rankings-{args.season}-{args.division}-{date_str}.log")
+    calculate_rankings(args.input, args.season, divisions, date_str, args.output)
```

Comment:
I don't see much advantage in having this as a separate function. When we have the script with `main` function, I would expect to have everything important written there so I don't have to look through the code to find the actual docstring. And you just redefine the parameters.

Reply:
I feel it separates concerns on an important level:
- main should only be run if one explicitly would like to run this script--the calculate rankings script. This takes care of acquiring the necessary values to run this script via the arguments
- the calculate_rankings function now does the actual calculation with the necessary parameters, regardless of where those parameters come from or how they were generated

This becomes much more relevant for something like the new workflow where we can then specifically call the `calculate_rankings` function with the appropriate parameters and not have to worry about unrelated, calculate_rankings-script-specific user-interface concerns. If anything, I think it makes the code more readable because it removes the user-interaction code from the actual logic and leaves a function that is dedicated to calculating rankings

Not a hill I'll die on, but I do feel this is a clearer and cleaner format. Happy to chat about it on a call if you'd like

Reply:
I personally prefer the function to do a single thing, which this approach violates as we are doing parsing, logging, calculating, saving, etc. in a single function. Main is the exception, as it is basically the execution of a script. I like the idea of a single unified parser defined elsewhere, though.

Reply:
I think we agree about ideas here and disagree about methods. I also find the function to be way too busy and doing too many things, and this is simply a first step in pulling out parsing. This pull request is more about adding a feature than about refactoring the code, and so I'm doing the small amount of refactoring that moves us towards single-purpose-functions and aligns with giving Ravi easier use